### PR TITLE
feat(Fieldset Header): Add Fieldset Header Component

### PR DESCRIPTION
**Pull request by deniaz**

### CODE REVIEW

File: src/compositions/FieldsetHeader/FieldsetHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    type?: Type;
+    disabled?: boolean;
+    bold?: boolean;
+    heading: string;
+    onClick?: () => void;
+    as?: keyof JSX.IntrinsicElements;
+};
+
+const renderType = (type: Type, id: string, size: Size, active: boolean, disabled: boolean) => {
+    if (type === Type.Switch) {
```
Comment (dwirz) [inline src/compositions/FieldsetHeader/FieldsetHeader.tsx:39]: why not using switch/case?

File: src/compositions/FieldsetHeader/FieldsetHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+export type FieldsetHeaderProps = {
+    size?: Size;
+    active?: boolean;
+    decorator?: ReactNode;
+    type?: Type;
+    disabled?: boolean;
+    bold?: boolean;
+    heading: string;
```
Comment (dwirz) [inline src/compositions/FieldsetHeader/FieldsetHeader.tsx:33]: didn't we talk about using `children: string`?

File: src/compositions/FieldsetHeader/FieldsetHeader.tsx (hunk 1 above)
Comment (deniaz) [reply src/compositions/FieldsetHeader/FieldsetHeader.tsx]: happy now? :smiling_face_with_three_hearts:

File: src/compositions/FieldsetHeader/FieldsetHeader.tsx (hunk 2 above)
Comment (deniaz) [reply src/compositions/FieldsetHeader/FieldsetHeader.tsx]: good point, my attention span is currently roughly 30s, so I forgot. :facepalm:

**Pull request merged**